sympy/polys/numberfields/subfield.py

```diff
@@ -93,40 +93,45 @@ def field_isomorphism_pslq(a, b):
         A = a.root.evalf(n)
         B = b.root.evalf(n)
 
-        basis = [1, B] + [ B**i for i in range(2, m) ] + [A]
```

Comment:
It is true, of course, that if `-a` is in `Q(b)` then so is `a` but I suspect that this line was the reason for the unexplained tests with `+/-a`. Clearly, `-a` should be used here to avoid confusion.

Reply:
True, I could see this having been overlooked, and then the cases further down were the result of trying to make it work.